### PR TITLE
Move block rate and log limit into toml config

### PR DESCRIPTION
**Pull request by ferglor**
https://smartcontract-it.atlassian.net/browse/AUTO-10027

In this PR, we are:

- Specifying defaults for the automation block rate and log limit in the toml config, based on the existing hard coded values [here](https://github.com/smartcontractkit/chainlink/blob/8863e3535f4c67f00fe6b380132bf72d78141215/core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/factory.go#L86)
- Cleaning up the old hard coded defaults
- Updating the oc2 config types in code to supply these new values

The configuration values are as follows:

## Block rate

- For arbitrum we specify a block rate of 2
- For all other supported chains, we specify a block rate of 1

## Log limit

- For arbitrum we specify a log limit of 1
- For eth we specify a log limit of 20
- For optimism, BSC, polygon, avax, and base we specify a log limit of 5
- For all other supported chains, we specify a log limit of 1

# Open questions

- Not sure if all the chains I've added defaults for are still supported - is there a list of currently supported chains?

### CODE REVIEW

**Review by ogtownsend — DISMISSED** [review]

File: core/chains/evm/config/toml/defaults/Polygon_Zkevm_Goerli.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Polygon_Zkevm_Goerli.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Polygon_Mumbai.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Polygon_Mumbai.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Optimism_Goerli.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Optimism_Goerli.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Ethereum_Rinkeby.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Ethereum_Rinkeby.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Ethereum_Goerli.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Ethereum_Goerli.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Base_Goerli.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Base_Goerli.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/Arbitrum_Goerli.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/Arbitrum_Goerli.toml:1]: this one is deprecated

File: core/chains/evm/config/toml/defaults/fallback.toml
Comment (FelixFan1992) [inline core/chains/evm/config/toml/defaults/fallback.toml:1]: can we also update for zksync and gnosis(xdai)?

**Review by infiloop2 — CHANGES_REQUESTED** [review]
We also have this config values through ocr offchain config which overrides these, not fully sure if these should within toml config also, will add to confusion. 

one way to reduce confusion could be to name these as "FallbackBlockRate" and "FallbackLogLimit", but we should also think through why hardcoded values are not sufficient

**Comment by ferglor** [comment]
> We also have this config values through ocr offchain config which overrides these, not fully sure if these should within toml config also, will add to confusion.
> 
> one way to reduce confusion could be to name these as "FallbackBlockRate" and "FallbackLogLimit", but we should also think through why hardcoded values are not sufficient

I'm happy to keep the hardcoded values (means we can just close this PR!), but my understanding was that we should have default values per chain set in the toml config, is that not the case?

**Comment by infiloop2** [comment]
keeping in toml might still be ok, but at least we should make it clear that these are only fallback values. I don't think we've had config yet which lives in both toml and offchain config yet so I would say discuss with @anirudhwarrier what would make sense here. (e.g. for new chains do we update toml or offchain config or both?)

**Comment by ferglor** [comment]
> keeping in toml might still be ok, but at least we should make it clear that these are only fallback values. I don't think we've had config yet which lives in both toml and offchain config yet so I would say discuss with @anirudhwarrier what would make sense here. (e.g. for new chains do we update toml or offchain config or both?)

Discussed with @anirudhwarrier and he's okay with scrapping this PR and relying on the defaults in code; is that okay with you @infiloop2 ?

**Comment by anirudhwarrier** [comment]
> keeping in toml might still be ok, but at least we should make it clear that these are only fallback values. I don't think we've had config yet which lives in both toml and offchain config yet so I would say discuss with @anirudhwarrier what would make sense here. (e.g. for new chains do we update toml or offchain config or both?)

After reconsidering this,  I think TOML config is overkill for a fallback config. I think this change should be discarded.

**Pull request closed without merging**